src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1789,6 +1789,7 @@ public void restartServer(boolean buildContainer) throws PluginExecutionExceptio
         if (generateFeatures) {
             //TODO: decide if we should do an optimized feature generation on a server restart
             // If we do not generate here, need to revisit compileDependenciesChanged section in DevMojo
+            // If we generate here, we need to skip install features if there is a failure
             optimizeGenerateFeatures();
```

Comment:
Do we need to skip install features if there is a failure generating features? There may still be user specified features that we would want to install.

Reply:
The expectation is that if we fail to generate features, it is due to conflicts somewhere and install feature will break anyway. We could investigate this further for the restart scenario (we are currently not skipping install).